config/v1/types_infrastructure.go

```diff
@@ -212,11 +214,14 @@ type PlatformStatus struct {
 // AWSServiceEndpoint store the configuration of a custom url to
 // override existing defaults of AWS Services.
 type AWSServiceEndpoint struct {
-	// Name is the name of the AWS service.
+	// name is the name of the AWS service.
+	// The list of all the service names can be found at https://docs.aws.amazon.com/general/latest/gr/aws-service-information.html
 	// This must be provided and cannot be empty.
+	//
+	// +kubebuilder:validation:Pattern=`^[a-z0-9-]+$`
 	Name string `json:"name"`
 
-	// URL is hostname only or fully qualified URI that overrides the default generated
+	// url is hostname only or fully qualified URI with schemes https or http, that overrides the default generated
 	// endpoint for a client.
 	// This must be provided and cannot be empty.
 	URL string `json:"url"`
```

Comment:
missing validation. You have a regex in the impl PR

Reply:
the valid and acceptable values are 
```
ec2.local
https://ec2.local
http//ec2.local
```

the impl doesn't have a regex, rather it adds `https://` if no scheme is set and then validates if http(s) schemes are set, If you have an regex that can do that let me know.. i don't think i have one that can validate URI+ hostname or http(s) correctly..